packages/core/src/utils/ErrorUtil.ts

```diff
@@ -22,4 +32,24 @@ const generateStackTrace = (): string => {
   return err.stack || "";
 };
 
-export {handleGlobalError, generateStackTrace};
+const isJSXError = (error: Error | UIError): error is UIError => {
+  return "componentStack" in error;
+};
+
+export const logIfComponentError = (
+  error: Error | UIError,
+  properties?: Properties,
+): Promise<boolean> => {
+  if (!isJSXError(error) || error.componentStack !== "") {
+    return createFalsePromise();
```

Comment:
I think we can move away from these helpers, I believe just `Promise.resolve(false)` is equivalent